src/v/bytes/details/io_allocation_size.h

```diff
@@ -29,28 +29,40 @@ class io_allocation_size {
     static constexpr size_t ss_max_small_allocation = 16384;
 
 public:
-    // >>> x=512
-    // >>> while x < int((1024*128)):
-    // ...     print(x)
-    // ...     x=int(((x*3)+1)/2)
-    // ...     x=int(min(1024*128,x))
-    // print(1024*128)
+    // This script computes the table immediately below, which are "ideal"
+    // allocation sizes: rounded up to the next true size supported by
+    // the seastar allocator. At <= 16K we apply the small pool indexing
+    // logic, and above we use powers of 2 since we are in the buddy allocator
+    // which only supports power-of-two sizes.
+    //
+    // We scale the target size by 1.47, i.e., 1.5 tweaked slightly to ensure
+    // we hit 16K at the small<->big pool boundary.
+    //
+    // def lg2(size: int):
+    //     return size.bit_length() - 1
+    // def p(v: object):
+    //     print(f"{v},")
+    // s = 512
+    // fb = 2 # idx_frac_bits
+    // while s <= 2**14:
+    //     idx = ((lg2(s) << fb) - ((1 << fb) - 1)) + ((s - 1) >> (lg2(s) - fb))
+    //     p((((1 << fb) | (idx & ((1 << fb) - 1))) << (idx >> fb)) >> fb)
+    //     s = int(s * 1.47)
+    // for e in [15, 16, 17]:
+    //     p(2**e)
     static constexpr auto alloc_table = std::to_array<uint32_t>(
-      // computed from a python script above
       {512,
        768,
-       1152,
-       1728,
-       2592,
-       3888,
-       5832,
-       8748,
-       13122,
-       19683,
-       29525,
-       44288,
-       66432,
-       99648,
+       1280,
+       1792,
+       2560,
+       3584,
+       6144,
+       8192,
+       12288,
+       16384,
+       32768,
+       65536,
        131072});
```

Comment:
The table is similar to the old one but "zero waste". At <= 16K, the differences are very small, i.e., the entries can be matched up 1:1 with only slight differences to avoid waste. 

Above 16K the new table doubles, since the allocator doubles, so sizes like 90K are just waste compared to 128K (i.e., the underlying alloc in both cases is 128K).

Reply:
I guess we could save even more waste by only using small pool sizes that divide pages cleanly but it's probably not worth it as the wastage-percent goes to zero once you seriously start using that size.

Reply:
@StephanDollberg that's only 2^n sizes, I think? Small pool always allocates spans, spans are always 2^n, and only power of 2 can exactly divide other powers of two (because the prime factorization of powers of 2 only have "2s", and so must anything that divides evenly into it).

So we couldn't do that and preserve 1.5x growth factor.

Reply:
The waste for non power of two can still be very small, however, and does vary across sizes.

Reply:
> that's only 2^n sizes

Yeah, don't think it's worth addressing as the wastage should be negligible.